tests/cpp/nodes/indexing/test_advanced.cpp

```diff
@@ -754,11 +813,34 @@ TEST_CASE("AdvancedIndexingNode") {
 
             graph.emplace_node<ArrayValidationNode>(adv);
 
-            THEN("We get the shape we expect") {
-                CHECK(adv->dynamic());
+            THEN("We get an Mx3 matrix") {
                 CHECK_THAT(adv->shape(), RangeEquals({-1, 3}));
-                // sizeinfo should be 1/3 * dyn_ptr size * 3 = dyn_ptr size
-                CHECK(adv->sizeinfo() == SizeInfo(dyn_ptr));
+
+                auto sizeinfo = adv->sizeinfo();
+
+                CHECK(sizeinfo.array_ptr == i_ptr);
+                CHECK(sizeinfo.multiplier == 3);
+                CHECK(sizeinfo.offset == 0);
+                CHECK(!sizeinfo.min);  // doesn't know unless it asks i
```

Comment:
Is this equivalent to `!sizeinfo.min.has_value()`? Or will it be true if `sizeinfo.min == 0`?

Reply:
The former, see https://en.cppreference.com/w/cpp/utility/optional/operator_bool.html